src/SPC/builder/unix/library/gettext.php

```diff
@@ -4,36 +4,34 @@
 
 namespace SPC\builder\unix\library;
 
+use SPC\util\executor\UnixAutoconfExecutor;
+
 trait gettext
 {
     protected function build(): void
     {
-        $extra = $this->builder->getLib('ncurses') ? ('--with-libncurses-prefix=' . BUILD_ROOT_PATH . ' ') : '';
-        $extra .= $this->builder->getLib('libxml2') ? ('--with-libxml2-prefix=' . BUILD_ROOT_PATH . ' ') : '';
-
-        $zts = $this->builder->getOption('enable-zts') ? '--enable-threads=isoc+posix ' : '--disable-threads ';
+        $autoconf = UnixAutoconfExecutor::create($this)
+            ->optionalLib('ncurses', "--with-libncurses-prefix={$this->getBuildRootPath()}")
+            ->optionalLib('libxml2', "--with-libxml2-prefix={$this->getBuildRootPath()}")
+            ->addConfigureArgs(
+                '--disable-java',
+                '--disable-c++',
+                '--with-included-gettext',
+                "--with-iconv-prefix={$this->getBuildRootPath()}",
+            );
 
-        $cflags = $this->builder->getOption('enable-zts') ? '-lpthread -D_REENTRANT' : '';
-        $ldflags = $this->builder->getOption('enable-zts') ? '-lpthread' : '';
+        // zts
+        if ($this->builder->getOption('enable-zts')) {
+            $autoconf->addConfigureArgs('--enable-threads=isoc+posix')
+                ->appendEnv([
+                    'CFLAGS' => '-lpthread -D_REENTRANT',
+                    'LDFLGAS' => '-lpthread',
+                ]);
+        } else {
+            $autoconf->addConfigureArgs('--disable-threads');
+        }
 
-        shell()->cd($this->source_dir)->initializeEnv($this)
-            ->appendEnv(['CFLAGS' => $cflags, 'LDFLAGS' => $ldflags])
-            ->exec(
-                './configure ' .
-                '--enable-static ' .
-                '--disable-shared ' .
-                '--enable-pic ' .
```

Comment:
We need to make sure to keep this everywhere. --with-pic too. Sadly there's no general standard. Or we just pass both by default and ignore the "unknown option" warnings?